econml/tree/_splitter.pxd

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+# This code is a fork from: https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/tree/_splitter.pxd
+# published under the following license and copyright:
+# BSD 3-Clause License
+#
+# Copyright (c) 2007-2020 The scikit-learn developers.
+# All rights reserved.
+
+# See _splitter.pyx for details.
+
+import numpy as np
+cimport numpy as np
+
+from ._criterion cimport Criterion
+
+from ._tree cimport DTYPE_t          # Type of X

```

Comment:
Yeah just in case we want to use different types in the future. sklearn actually uses float32 for X for better memory usage. We could do the same or not. Wanted to have the option in the future.